src/analysis/convex/extreme.lean

```diff
@@ -175,13 +168,61 @@ subset.antisymm (λ x hx, ⟨hx.1, hAB.extreme_points_subset_extreme_points hx
 end has_smul
 
 section ordered_semiring
-variables {𝕜} [ordered_semiring 𝕜] [add_comm_group E] [module 𝕜 E] {A B : set E} {x : E}
+variables [ordered_semiring 𝕜] [add_comm_group E] [add_comm_group F] [Π i, add_comm_group (π i)]
+  [module 𝕜 E] [module 𝕜 F] [Π i, module 𝕜 (π i)] {A B : set E} {x : E}
 
 lemma is_extreme.convex_diff (hA : convex 𝕜 A) (hAB : is_extreme 𝕜 A B) :
   convex 𝕜 (A \ B) :=
 convex_iff_open_segment_subset.2 (λ x₁ ⟨hx₁A, hx₁B⟩ x₂ ⟨hx₂A, hx₂B⟩ x hx,
     ⟨hA.open_segment_subset hx₁A hx₂A hx, λ hxB, hx₁B (hAB.2 hx₁A hx₂A hxB hx).1⟩)
 
+@[simp] lemma extreme_points_prod (s : set E) (t : set F) :
+  (s ×ˢ t).extreme_points 𝕜 = s.extreme_points 𝕜 ×ˢ t.extreme_points 𝕜 :=
+begin
+  ext,
+  refine (and_congr_right $ λ hx, ⟨λ h, _, λ h, _⟩).trans (and_and_and_comm _ _ _ _),
+  split,
+  { rintro x₁ hx₁ x₂ hx₂ ⟨a, b, ha, hb, hab, hx'⟩,
+    refine (h (mk_mem_prod hx₁ hx.2) (mk_mem_prod hx₂ hx.2) _).imp
+      (congr_arg prod.fst) (congr_arg prod.fst),
+    refine ⟨a, b, ha, hb, hab, prod.ext hx' _⟩,
+    simp_rw [prod.smul_mk, prod.mk_add_mk, convex.combo_self hab] },
+  { rintro x₁ hx₁ x₂ hx₂ ⟨a, b, ha, hb, hab, hx'⟩,
+    refine (h (mk_mem_prod hx.1 hx₁) (mk_mem_prod hx.1 hx₂) _).imp
+      (congr_arg prod.snd) (congr_arg prod.snd),
+    refine ⟨a, b, ha, hb, hab, prod.ext _ hx'⟩,
+    simp_rw [prod.smul_mk, prod.mk_add_mk, convex.combo_self hab] },
```

Comment:
This is the golf I was thinking of:
```suggestion
  { rintro x₁ hx₁ x₂ hx₂ hx_fst,
    refine (h (mk_mem_prod hx₁ hx.2) (mk_mem_prod hx₂ hx.2) _).imp
      (congr_arg prod.fst) (congr_arg prod.fst),
    rw ← prod.image_mk_open_segment_left,
    exact ⟨_, hx_fst, prod.mk.eta⟩, },
  { rintro x₁ hx₁ x₂ hx₂ hx_snd,
    refine (h (mk_mem_prod hx.1 hx₁) (mk_mem_prod hx.1 hx₂) _).imp
      (congr_arg prod.snd) (congr_arg prod.snd),
    rw ← prod.image_mk_open_segment_right,
    exact ⟨_, hx_snd, prod.mk.eta⟩, },
```

Reply:
Hardly a golf at this point, but okay :P